src/wallet/wallet.cpp

```diff
@@ -1885,7 +1885,7 @@ std::optional<ZTXOSelector> CWallet::ZTXOSelectorForAddress(
         const libzcash::PaymentAddress& addr,
         bool requireSpendingKey,
         TransparentCoinbasePolicy transparentCoinbasePolicy,
-        bool allowAddressLinkability) const
+        std::optional<UnifiedAccountSpendingPolicy> spendingPolicy) const
```

Comment:
It is not documented what `std::nullopt` means here. In the context of this PR we can see that it replaces `allowAddressLinkability=false`, but that won't be clear from just reading the code after the PR. Perhaps it would be better to add `DoNotAllowAddressLinkability` (or some better name) as an alternative in `UnifiedAccountSpendingPolicy`, rather than using a `std::optional`?

Reply:
It’s [documented in the .h](https://github.com/zcash/zcash/pull/6457/files/df383abab2897f06193bb64dc47d25a8a1e15c75#diff-9ce137cd784ea308778842120aa2af6d2bb8369485b71f25e72b2a32cf0a5b21R1583-R1586)

`std::nullopt` is similar to the old `false`, but`UnifiedAccountSpendingPolicy::ShieldedWithSingleTransparentAddress` also behaves like `false` in the case where the provided UA has a transparent receiver. Your `DoNotAllowLinkingAddresses` is partitioned into `UnifiedAccountSpendingPolicy::ShieldedWithSingleTransparentAddress` and `UnifiedAccountSpendingPolicy::ShieldedOnly`. `std::nullopt` here isn’t about not linking addresses, but it means “don’t try to map this UA to an account”.

I generally prefer an explicit `std::optional` to make it clear that no-value is distinct from the set of values. In particular, the upcomig `z_sendfromaccount` uses a non-optional `UnifiedAccountSpendingPolicy` (since there’s no way to not have an account in that case).